pkg/events/events.go

```diff
@@ -91,23 +91,22 @@ type RegisterVendorEvent struct {
 
 // VendorClaimEvent event
 type VendorClaimEvent struct {
-	VendorIdentifier Identifier    `json:"vendorIdentifier"`
-	OrgIdentifier    Identifier    `json:"orgIdentifier"`
-	OrgName          string        `json:"orgName"`
+	VendorIdentifier Identifier `json:"vendorIdentifier"`
+	OrgIdentifier    Identifier `json:"orgIdentifier"`
+	OrgName          string     `json:"orgName"`
 	// OrgKeys is a list of JWKs which are used to
 	// 1. encrypt data to be decrypted by the organization,
 	// 2. sign consent JWTs,
 	// 3. sign organization related events (e.g. endpoint registration).
-	OrgKeys          []interface{} `json:"orgKeys,omitempty"`
-	Start            time.Time     `json:"start"`
-	End              *time.Time    `json:"end,omitempty"`
+	OrgKeys []interface{} `json:"orgKeys,omitempty"`
+	Start   time.Time     `json:"start"`
+	End     *time.Time    `json:"end,omitempty"`
 }
 
 type jsonEvent struct {
-	EventType     string                 `json:"type"`
-	EventIssuedAt time.Time              `json:"issuedAt"`
-	EventPayload  map[string]interface{} `json:"payload"`
-	data          []byte
+	EventType     string      `json:"type"`
+	EventIssuedAt *time.Time  `json:"issuedAt,omitempty"`
```

Comment:
Why is EventIssuedAt optionally?

Reply:
Because in the current on-disk JSON event storage, it's derived from the file name and set after deserializing it. If we also added it inside the file, there would be 2 sources of truth, the file name and whatever is specified inside it. In the domain model (Event interface) it's mandatory.

When we move to a distributed registry or move to storage in database, I believe this disk-based format will be removed anyways.

Reply:
So, in practice the EventIssuedAt is always set, be it Now(), `0001-01-01 00:00:00 +0000 UTC` or the actual time of the event. So, why not just make it not-optional? Than you can get rid of the Getter function.

Reply:
Fixed by moving some complexity to the corresponding unit test. EventIssuedAt is now mandatory. The getter function has to stay though, since the Event is an interface used to access the underlying struct (jsonEvent for instance).